From: Colin Ian King <colin.king@canonical.com>
Date: Fri, 29 Sep 2017 21:13:08 +0100
Subject: RDMA/hns: return 0 rather than return a garbage status value
Patch-mainline: v4.15-rc1
Git-commit: 63ea641f496ffe08b7d81589a1e1e1c43a6af9bd
References: bsc#1104427 FATE#326416

For the case where hr_qp->state == IB_QPS_RESET, an uninitialized
value in ret is being returned by function hns_roce_v2_query_qp.
Fix this by setting ret to 0 for this specific return condition.

Detected by CoverityScan, CID#1457203 ("Unitialized scalar variable")

Signed-off-by: Colin Ian King <colin.king@canonical.com>
Acked-by: Wei Hu (Xavier) <xavier.huwei@huawei.com>
Signed-off-by: Doug Ledford <dledford@redhat.com>
Acked-by: Thomas Bogendoerfer <tbogendoerfer@suse.de>
---
 drivers/infiniband/hw/hns/hns_roce_hw_v2.c |    1 +
 1 file changed, 1 insertion(+)

--- a/drivers/infiniband/hw/hns/hns_roce_hw_v2.c
+++ b/drivers/infiniband/hw/hns/hns_roce_hw_v2.c
@@ -2807,6 +2807,7 @@ static int hns_roce_v2_query_qp(struct i
 
 	if (hr_qp->state == IB_QPS_RESET) {
 		qp_attr->qp_state = IB_QPS_RESET;
+		ret = 0;
 		goto done;
 	}
 
